soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1532,73 +1894,33 @@ s16 Randomizer::GetItemFromGet(RandomizerGet randoGet, GetItemID ogItemId) {
             return GI_RUPEE_PURPLE;
         case RG_HUGE_RUPEE:
             return GI_RUPEE_GOLD;
-
         case RG_PIECE_OF_HEART:
             return GI_HEART_PIECE;
         case RG_HEART_CONTAINER:
-            // todo figure out what GI_HEART_CONTAINER_2 is
             return GI_HEART_CONTAINER;
-        case RG_MILK:
-            return GI_MILK; //todo logic around needing a bottle?
-
-        case RG_BOMBS_5:
-            return CUR_UPG_VALUE(UPG_BOMB_BAG) ? GI_BOMBS_5 : GI_RUPEE_BLUE;
-        case RG_BOMBS_10:
-            return CUR_UPG_VALUE(UPG_BOMB_BAG) ? GI_BOMBS_10 : GI_RUPEE_BLUE;
-        case RG_BOMBS_20:
-            return CUR_UPG_VALUE(UPG_BOMB_BAG) ? GI_BOMBS_20 : GI_RUPEE_BLUE;
-
-        case RG_BOMBCHU_5:
-            return GI_BOMBCHUS_5;
-        case RG_BOMBCHU_10:
-            return GI_BOMBCHUS_10;
-        case RG_BOMBCHU_20:
-            return GI_BOMBCHUS_20;
-        case RG_BOMBCHU_DROP:
-            return GI_BOMBCHUS_5; //todo figure out what we want to do for chu drops
-
-        case RG_ARROWS_5:
-            return CUR_UPG_VALUE(UPG_QUIVER) ? GI_ARROWS_SMALL : GI_RUPEE_BLUE;
-        case RG_ARROWS_10:
-            return CUR_UPG_VALUE(UPG_QUIVER) ? GI_ARROWS_MEDIUM : GI_RUPEE_BLUE;
-        case RG_ARROWS_30:
-            return CUR_UPG_VALUE(UPG_QUIVER) ? GI_ARROWS_LARGE : GI_RUPEE_BLUE;
-
+        
         case RG_DEKU_NUTS_5:
+        case RG_BUY_DEKU_NUT_5:
             return GI_NUTS_5;
         case RG_DEKU_NUTS_10:
+        case RG_BUY_DEKU_NUT_10:
             return GI_NUTS_10;
-
-        case RG_DEKU_SEEDS_30:
-            return CUR_UPG_VALUE(UPG_BULLET_BAG) ? GI_SEEDS_30 : GI_RUPEE_BLUE;
-
         case RG_DEKU_STICK_1:
+        case RG_BUY_DEKU_STICK_1:
             return GI_STICKS_1;
-
-        // RANDOTODO these won't be used until we implement shopsanity/scrub shuffle
-        case RG_RED_POTION_REFILL:
-        case RG_GREEN_POTION_REFILL:
-        case RG_BLUE_POTION_REFILL:
-            return GI_NONE;
-
+        case RG_TREASURE_GAME_SMALL_KEY:
+            return GI_DOOR_KEY;
         case RG_TREASURE_GAME_HEART:
             return GI_HEART_PIECE_WIN;
         case RG_TREASURE_GAME_GREEN_RUPEE:
-            return GI_RUPEE_GREEN_LOSE; //todo figure out how this works outside of the game
-
-        case RG_TRIFORCE:
-            return GI_RUPEE_BLUE; //todo figure out what this is/does
-
-        case RG_HINT:
-            return GI_RUPEE_BLUE; //todo
-
+            return GI_RUPEE_GREEN_LOSE;
         default:
             if (!IsItemVanilla(randoGet)) {
-                return randoGet;
+                return (GetItemID)randoGet;
```

Comment:
This cast here is why had this function returning an `int` instead of a `GetItemID`, since this function could return a `GetItemID` or a `RandomizerGet`. I'm OK with keeping it if returning an enum type is preferred, but I wonder if there is a data type we could use that could clearly encapsulate both without adding too much weird syntax. Like a `union` or something similar. But maybe that's out of scope for this PR?

Reply:
i'd consider that out of scope for this PR, but definitely worth looking into. maybe make an issue and add it to the v3 milestone?

Reply:
I had kinda been wondering if unions existed in c/c++, I briefly looked at them and they looked quite a bit different than what I’m used to in typescript